ckanext/opendatani/controller.py

```diff
@@ -249,3 +253,32 @@ def resource_read(self, id, resource_id):
                     h.flash_error('Sorry, this file is too large to be able to display in the browser, please download the data resource to examine it further.')
         template = self._resource_template(dataset_type)
         return render(template, extra_vars=vars)
+
+    def retrieve_report(self, org):
+
+        context = {'model': model, 'session': model.Session,
+                   'user': toolkit.c.user,
+                   'auth_user_obj': toolkit.c.userobj}
+        data_dict = {'id': toolkit.request.params.get('user')}
+
+        try:
+            toolkit.check_access('sysadmin', context)
+        except toolkit.NotAuthorized:
+            toolkit.abort(401,
+                          toolkit._('You are not authorized to access this report. Please login to try again.'))
+
+        data_dict = {'q': 'organization:{0}'.format(org)}
+        resource_data = toolkit.get_action('report_resources_by_organization')({}, data_dict)
+
+        FIELDS = resource_data[0].keys()
+        output = StringIO()
+        writer = csv.DictWriter(output, fieldnames=FIELDS, quoting=csv.QUOTE_MINIMAL)
+        writer.writeheader()
+        yield output.getvalue()
+
+        for data in resource_data:
+            output = StringIO()
+            writer = csv.DictWriter(output, fieldnames=FIELDS, quoting=csv.QUOTE_MINIMAL)
+            writer.writerow(data)
+            yield '<br/>' + output.getvalue()
```

Comment:
This is just displaying a text in a new window. you need to save that text (without `<br/>`) in some dummy file and remove after download. Like it's done here https://github.com/CivicActions/ckanext-ed/blob/master/ckanext/ed/controller.py#L162-L182